daemon/analytics.c

```diff
@@ -335,6 +357,84 @@ void analytics_alarms_notifications(void)
     buffer_free(b);
 }
 
+/*
+ * Checks for the existance of .install_type file and reads it
+ */
+void analytics_get_install_type(void)
+{
+    char *install_type_filename;
+    analytics_set_data_str(&analytics_data.netdata_install_type, "");
+    analytics_set_data_str(&analytics_data.netdata_prebuilt_distro, "");
+
+    install_type_filename =
+        mallocz(sizeof(char) * (strlen(netdata_configured_user_config_dir) + strlen(".install-type") + 2));
+    sprintf(install_type_filename, "%s/%s", netdata_configured_user_config_dir, ".install-type");
```

Comment:
Please, use `snprintfz` instead `sprintf`.

Reply:
Hmm, thanks Thiago. In that case though, it would be better to set `install_type_filename` to a fixed length (maybe `FILENAME_MAX`) and use that then in `snprintfz`. We are also pretty much sure that the `sprintf` won't exceed the length of `install_type_filename`, so it should be safe this way too? It's also the way I have used in most of the cases here...

If it's better to use a static length in such files related variables, I could change it to like that in all cases...

Reply:
Or like this: https://github.com/netdata/netdata/pull/11237/commits/f03782563f5f426ef77da0102cb314a89c20516b if you're ok. :-)

Reply:
I agree that for this case we cannot exceed the size, my suggestion to change the function was because the warning we have in the official documentation

```
Because sprintf() and vsprintf() assume an arbitrarily long string, callers must be careful not to overflow the actual space; this is often impossible to assure.  Note that the length of the strings produced  is  locale-de‐
       pendent and difficult to predict.  Use snprintf() and vsnprintf() instead (or asprintf(3) and vasprintf(3)).

       Code such as printf(foo); often indicates a bug, since foo may contain a % character.  If foo comes from untrusted user input, it may contain %n, causing the printf() call to write to memory and creating a security hole.
``` 

Thanks this warning, `snprintf` is always the safest option. 
I am OK with the suggested commit. :)